### PR TITLE
feat(installer): redesign utility options

### PR DESCRIPTION
**Pull request by validcube**
## ✨ Redesign utility options

This replace the 3 dots menu with buttons, improve visibility

> Save APK
> Share Log

This also removed the share option from the installer

image

### CODE REVIEW

**Comment by Ushie** [comment]
What'd you do to share?

File: lib/ui/views/installer/installer_viewmodel.dart [hunk 1]
```diff
@@ -264,17 +254,14 @@ class InstallerViewModel extends BaseViewModel {
...
     switch (value) {
       case 0:
-        shareResult();
+        exportResult();
         break;
       case 1:
         exportResult();
         break;
-      case 2:
-        shareLog();
```
Comment (validcube) [inline lib/ui/views/installer/installer_viewmodel.dart:276]: bruh

**Comment by Ushie** [comment]
Use contained tonal icon buttons https://m3.material.io/components/icon-buttons/overview

**Comment by Ushie** [comment]
Also @PalmDevs could you take a look at this?

**Comment by PalmDevs** [comment]
I'd personally suggest using the already-existing [Bottom App Bar](https://m3.material.io/components/bottom-app-bar/overview) component.

**Comment by Ushie** [comment]
I'd completely forgotten about that, looks perfect

**Comment by ILoveOpenSourceApplications** [comment]
#886 can be closed if this is implemented right?

**Review by oSumAtrIX — APPROVED** [review]

**Comment by Ushie** [comment]
> #886 can be closed if this is implemented right?

Yes

**Review by Ushie — CHANGES_REQUESTED** [review]
Use [Bottom App Bar](https://m3.material.io/components/bottom-app-bar/overview) for the patch result options

**Comment by Ushie** [comment]
@PalmDevs Need a bit of help, what should the copy logs button do?:
- Copy logs as text
- Copy file uri with the logs as the content
- Open share menu (current behavior)


image

**Comment by Ushie** [comment]
Also, should the Install FAB contain a label? or should it be Icon only

**Comment by PalmDevs** [comment]
> @PalmDevs Need a bit of help, what should the copy logs button do?:
> - Copy logs as text
> - Copy file uri with the logs as the content
> - Open share menu (current behavior)
> 
> [Screenshot]

I think the best choice would be to open a file picker to let the user save the logs. The user (if they want to share) can access their file manager and share it from there.

**Comment by PalmDevs** [comment]
The icons should also be changed. The icon for exporting the APK should be "Save" (floppy disk icon), and the icon for saving log should be "Post add" icon. References are found on Google's Icons website.

**Comment by Ushie** [comment]
image

**Comment by ILoveOpenSourceApplications** [comment]
> image

This looks good but it would look even better if the style is similar to that of how bottom app bar is in Gmail app. When scrolling down the container disappears and when scrolled up the container reappears. Also the installation button can be similar to the compose button in Gmail app. When still will show the icon and "Install" text next to it and scrolling down will shrink it to just the icon.

Demo:

https://github.com/ReVanced/revanced-manager/assets/117499019/402975df-154a-4066-81c6-6cd09f0d704f

**Comment by Ushie** [comment]
The FAB in BottomAppBar can't be shown separetely, its part of the BottomAppBar, if the BottomAppBar gets hidden then the FAB does too

**Comment by ILoveOpenSourceApplications** [comment]
Under scrolling in guidelines,  my opinion is represented and can be seen in action. Please have a look at it. This FAB implementation can also benefit throughout the UI. When clicking the "Storage" button, "Patch" button and now "Install" button.

**Comment by Ushie** [comment]
I looked into hiding on scroll, and I think the implementation is rather annoying just for something I personally think has no use in this context

Maybe if @validcube wants to add hide on scroll, they can but I personally wont (I dont know flutter enough to do it in a way I think is good code)

As for using Tonal IconButtons, I tried both standard and tonal versions and I prefer tonal, I don't see specs that speak against using it

**Pull request merged**